powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/RecoveryCodeActivationRequest.java

```diff
@@ -28,22 +31,56 @@
  */
 @Data
 public class RecoveryCodeActivationRequest {
+
+    @Schema(description = "Whether recovery codes should be generated")
     private Boolean generateRecoveryCodes;
+
+    @Schema(description = "Recovery code value used for activation")
+    @NotBlank(message = "Recovery code value must not be empty when creating activation using recovery code")
     @ToString.Exclude
     private String recoveryCode;
+
+    @Schema(description = "Recovery PUK value used for activation")
+    @NotBlank(message = "Recovery PUK value must not be empty when creating activation using recovery code")
     @ToString.Exclude
     private String puk;
+
+    @Schema(description = "Application key")
+    @NotBlank(message = "Application key must not be empty when creating activation using recovery code")
     private String applicationKey;
+
+    @Schema(description = "Maximum number of failures for the activation")
+    @Positive(message = "Maximum failure count must be positive when creating activation using recovery code")
     private Long maxFailureCount;
+
+    @Schema(description = "Identifier of the temporary key for encryption")
     private String temporaryKeyId;
+    @Schema(description = "Ephemeral public key used in encryption")
+
+    @NotBlank(message = "Ephemeral public key must not be empty when creating activation using recovery code")
     private String ephemeralPublicKey;
+
+    @Schema(description = "Encrypted data used in encryption")
```

Comment:
The second part of the sentence doesn't sound right.
```suggestion
    @Schema(description = "Encrypted data")
```